app/controllers/claims_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ClaimsController < ApplicationController
+  def new
+    contribution = Listing.find(params[:contribution_id])
+    if contribution.person&.email.blank?
+      flash.now[:alert] = "We are sorry, this contributor hasn't provided
+                         an email address yet and can't communicate this way".squish
+    end
+    render locals: { contribution: contribution }
+  end
+
+  def create
+    # TODO: Need to handle race conditions to prevent creating multiple matches for same contribution.
+
+    contribution = Listing.find(params[:contribution_id])
+    ActiveRecord::Base.transaction do
+      if current_user.person.blank?
+        Person.create_from_peer_to_peer_params!(current_user, name: peer_to_peer_match_params[:peer_alias],
+                                                preferred_contact_method_id: peer_to_peer_match_params[:preferred_contact_method_id],
+                                                contact_info: peer_to_peer_match_params[:preferred_contact_details])
+      end
+      Match.create_match_for_contribution!(contribution, current_user)
+      contribution.matched!
+    end
+    notify_peer_and_log_communication!(contribution)
+    redirect_to contribution_path(params[:contribution_id]), notice: 'Claim was successful!'
+  rescue
+    flash.now[:alert] = 'There was an error. Please try again.'
+    render :new, locals: { contribution: contribution }
+  end
```

Comment:
I think i'd lean towards letting this fail fast and relying instead on a global error handler (existing [issue](https://github.com/rubyforgood/mutual-aid/issues/734)).